src/common/os_linux.c

```diff
@@ -144,6 +147,40 @@ os_mkstemp(char *temp)
 int
 os_posix_fallocate(int fd, os_off_t offset, off_t len)
 {
+
+#ifdef __FreeBSD__
```

Comment:
Ifdef freebsd in os_linux.c looks a bit silly ;). Not necessarily in this PR, but I think we should replace _linux with _unix in all file names.